doc/build/dts/bindings.rst

```diff
@@ -1011,6 +1011,12 @@ General rules
 
   Do not use any other style for long or multi-line strings.
 
+- Do not use uppercase letters (``A`` through ``Z``) or underscores (``_``) in
+  property names. Use lowercase letters (``a`` through ``z``) instead of
+  uppercase. Use dashes (``-``) instead of underscores. (The one exception to
+  this rule is if you are replicating a well-established binding from somewhere
+  like Linux.)
```

Comment:
Should we add a comment here about why this is the case?  `This is due to how Zephyr encodes devicetree properties`

Reply:
> `This is due to how Zephyr encodes devicetree properties`

It's not, though -- it's just a general recommendation to follow the pattern established by DT spec.